BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb

```diff
@@ -350,18 +350,43 @@ class CbrainTask::<%= name %> < <%= descriptor['cbrain:inherits-from-class'] ||
     end
 
 %   end
+    # Helper for checking whether outputs are optional
+%   optionalFiles = outputs.select { |o| o['optional'] }
+    isOptional = lambda do |f| 
+      [<%= optionalFiles.map{ |x| ":'"+x['id']+"'" }.join(",") %>].include?(f) 
+    end
+
     # Save (and register) all generated files to the results data provider
     outputs.each do |param, paths|
       paths = [paths] unless paths.is_a?(Enumerable)
       paths.each do |path|
-        next unless path.present? && File.exists?(path)
-
-        self.addlog("Saving result file #{path}")
-        name = File.basename(path)
         
-	userfile_class = Userfile.suggested_file_type(name)
-        userfile_class ||= ( File.directory?(path) ? FileCollection : SingleFile )
+        # Print a warning if the file is not present
+        # Note that, since the output may be optional, the process may still be successful
+        unless path.present? && File.exists?(path)
+          self.addlog("Unable to find optional output file: #{path}") if isOptional.(param)
+          next
+        end
```

Comment:
I think it is supposed to print only if the file is optional (because it prints a warning about required files above), but  the "next" statement is supposed to run in either case